File: Expenses.Core.Entities/ProductPurchase.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace Expenses.Core.Entities
{
    public class ProductPurchase
    {
        [Key]
        public int Id { get; set; }

        public ProductDetails ProductDetail { get; set; }
        public int? ProductDetailId { get; set; }

        [Required]
        public Purchase Purchase { get; set;}
        public int PurchaseId { get; set; }

        [Required]
        public Product Product { get; set; }
        public int ProductId { get; set; }

        [Required]
        public double Price { get; set; }
        public int? Quantity { get; set; }
        public double? Weight { get; set; }
    }
}

File: Expenses.UI/ClientApp/src/app/models/productPurchase.ts
import { Product } from "./product";
import { ProductDetails } from "./productDetails";
import { Purchase } from "./purchase";

export interface ProductPurchase {
    id: number;
    purchaseId: number;
    purchase: Purchase;
    productDetail: ProductDetails;
    productId: number;
    product: Product;
    price: number;
    quantity: number;
    weight: number;
    details: string;
  }